### PR TITLE
Add prove server mode

### PR DESCRIPTION
**Pull request by noel2004**
An implement for the server mode according to #19, also:

+ Add dependencies for server: tokio/tonic/tonic-build and the "server" feature

+ Add build script for the protobuf definition

+ Script for integration test

*P.S: there seems to be too many println which output to console for a server, should we remove some of them later?*

### CODE REVIEW

**Comment by noel2004** [comment]
The response of server for prove request is like this:

```javascript

{
  is_valid: true,
  time_cost: 2.6957576,
  proof: [
    '17013270398757165371163109406711653183170826015429694925667943800852466964226',
    '1035406126346085907060746804964340529136159583191466579826078572068208188971',
    '2430762324238287556500195466694649142422614896199476916609236874761733944783',
    '10648591426529877002471018408132203239398961701371078197617944392535821484394',
    '20463836388850053222297346881167621438516540294983141151801113840218191201849',
    '20167246415138357540617881736851406439275063557410642323985959862312214900435',
    '13616567045872191463460367057789470862545695429861353286118008929292855976191',
    '982548387243388125047530655920888804797560549650777880201994316146718610288',
    '1871646330332914786233925277217892520871390609921353709469897122420735505948',
    '5663125027789153905219185918109934458043425832564600311082883871072947898895',
    '2355661186967715275224592933058486475975899043168973143552429408934287704903',
    '7552432350856564481442980125756403122928440227797715782622574147969028007765',
    '21855372145391004883567167700702245400633468177593122168203192699958310559357',
    '20034523632999922919336074104206570911502646901791511196581013845865541957599',
    '14501625223551583455647792819076342617188182939598132706432661372782919035597',
    '9723275502636929148034013462462682007704227780154869885597931129361691354111',
    '12332992311479196858922979507623837489987339475967357382641517977105494422690',
    '12617526464949170273144596777981369277373492157892804504861045806358812324078',
    '1619602872459345229093121281635920694395472601986158081308929138491859384506',
    '19043735244606990532016792673264595622107198789596798813609956310583319222359',
    '9907074293315160954592026713319550502805609589491009103665289715099983968568',
    '19377285646440772113201809237117031436064409044461324035265737486024730264792',
    '18221588417733051322062091756674149464691985302343300660351242654401132166759',
    '7036219056848959911331408600514065664949303327407534547995225124959536988901',
    '1006755320468553820801330365477422767916475102262043003556633468704926966255',
    '14555112594070898369993201732540597993515578179572596404493678371119135699406',
    '6140387524638844327355700514880017863987288867824234383838489385798015601206',
    '1214796337028854972001151792438719391147364447492597693448733847620106056221',
    '6023335038353415601883584371594876415015918601459908751661990099070346505588',
    '18170784767323215310054715227277440809538868346973508164099395278383019477107',
    '3734927252677315941698977615250099327621133945598870484589421844354090181756',
    '6936246054474619407126268803548119113515096368163095957145080953754207425136',
    '17129129219633997023313889042792638184552903006882409068920919977620386601032'
  ],
  inputs: [
    '21867347236198497199818917118739170715216974132230970409806500217655788551452',
    '3',
    '4'
  ]
}


```

**Comment by HAOYUatHZ** [comment]
> P.S: there seems to be too many println which output to console for a server, should we remove some of them later?

Not an emergency. But you are right, we can use log::info, log::debug, log::warn to control the logging level.

File: src/main.rs [hunk 1]
```diff
@@ -221,6 +246,76 @@ fn dump_lagrange(opts: DumpLagrangeOpts) {
...
+            let perf_t = std::time::SystemTime::now();
+
+            let mut circut = circuit_base.clone();
+            let witness_ret = reader::load_witness_from_array::<Bn256>(witness);
+            if let Ok(witness) = witness_ret {
+                circut.witness = Some(witness);
+            } else {
+                return server::CoreResult::any_prove_error(witness_ret, validate_only);
+            }
+            let proof_ret = setup.prove(circut);
```
Comment (HAOYUatHZ) [inline src/main.rs:282]: Here it actually proves  every time, and proving is very time-consuming.

we'd better add a conditional branch for https://github.com/Fluidex/plonkit/blob/006e688997eba2bd3687505999a213cf1eb93336/src/main.rs#L260-L270

Comment (HAOYUatHZ) [reply src/main.rs:282]: Ignore me. I find that only **prove** is requsted from the spec.

Comment (HAOYUatHZ) [reply src/main.rs:282]: Ignore me. I find that only **prove** is requsted from the spec.

**Pull request merged**